public/app/plugins/panel/graph/Legend/LegendSeriesItem.tsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { TimeSeries } from 'app/core/core';
+import withColorPicker from 'app/core/components/colorpicker/withColorPicker';
+
+export const LEGEND_STATS = ['min', 'max', 'avg', 'current', 'total'];
+
+export interface LegendLabelProps {
+  series: TimeSeries;
+  asTable?: boolean;
+  hidden?: boolean;
+  onLabelClick?: (series, event) => void;
+  onColorChange?: (series, color: string) => void;
+  onToggleAxis?: (series) => void;
+}
+
+export interface LegendValuesProps {
+  values?: boolean;
+  min?: boolean;
+  max?: boolean;
+  avg?: boolean;
+  current?: boolean;
+  total?: boolean;
+}
+
+type LegendItemProps = LegendLabelProps & LegendValuesProps;
+
+interface LegendItemState {
+  yaxis: number;
+}
+
+export class LegendItem extends React.PureComponent<LegendItemProps, LegendItemState> {
```

Comment:
Same as with `Legend.tsx`, would be awesome if this file was split into separate files to increase readability.